### PR TITLE
[glsl-in] Implement swizzle for r-values

### PR DESCRIPTION
**Pull request by pjoe**
Related to #210

### CODE REVIEW

File: src/front/glsl/variables.rs [hunk 1]
```diff
@@ -96,4 +97,88 @@ impl Program {
...
     }
+
+    pub fn field_selection(
+        &mut self,
+        expression: Handle<Expression>,
+        name: &str,
+        meta: TokenMetadata,
+    ) -> Result<Handle<Expression>, ErrorKind> {
+        let type_inner = self.resolve_type(expression)?;
+        match type_inner {
```
Comment (kvark) [inline src/front/glsl/variables.rs:108]: let's match on the value always, i.e. `match *self.resolve_type(expression)? {..}`

Comment (pjoe) [reply src/front/glsl/variables.rs]: Done, except borrow checker wouldn't let me deref with `*` (something about unable to move out of shared ref), so had to settle for `match self.resolve_type(expression)? {..}`

Comment (kvark) [reply src/front/glsl/variables.rs]: The whole point of borrowing by value is that it lets you see which things *are* borrowed, and which not.
So, you see that the borrow checker complains about `members` field being moved out? Mark it as `ref members`.

Comment (pjoe) [reply src/front/glsl/variables.rs]: Ahhh, maybe one day I will learn to actually listen to what the borrow checker tells me :)

**Review by kvark — APPROVED** [review]
Thank you!

**Pull request merged**